src/qililab/pulse/qblox_compiler.py

```diff
@@ -0,0 +1,386 @@
+        pair = ([float(w) for w in sequencer.weights_i], [float(w) for w in sequencer.weights_q])
+        if (sequencer.path_i, sequencer.path_q) == (1, 0):
+            pair = pair[::-1]  # swap paths
+        weights.add_pair(pair=pair, indices=(0, 1))
+        return weights
+
+    def _append_acquire_instruction(
+        self,
+        loop: Loop,
+        bin_index: Register | int,
+        sequencer: AWGQbloxADCSequencer,
+        weight_regs: tuple[Register, Register],
+    ):
+        """Append an acquire instruction to the loop."""
+        weighed_acq = sequencer.weighed_acq_enabled
+        qblox_module = self._get_instrument_from_sequencer(sequencer)
+
+        acq_instruction = (
+            AcquireWeighed(
+                acq_index=0,
+                bin_index=bin_index,
+                weight_index_0=weight_regs[0],
+                weight_index_1=weight_regs[1],
+                wait_time=qblox_module._MIN_WAIT_TIME,  # pylint: disable=protected-access
+            )
+            if weighed_acq
+            else Acquire(
+                acq_index=0,
+                bin_index=bin_index,
+                wait_time=qblox_module._MIN_WAIT_TIME,  # pylint: disable=protected-access
+            )
+        )
+        loop.append_component(acq_instruction)
+
+    def _init_weights_registers(self, registers: tuple[Register, Register], program: Program):
+        """Initialize the weights `registers` and place the required instructions in the setup block of the `program`."""
+        move_0 = Move(0, registers[0])
+        move_1 = Move(1, registers[1])
+        setup_block = program.get_block(name="setup")
+        setup_block.append_components([move_0, move_1], bot_position=1)
+
+    def get_sequencer_schedule(
+        self, pulse_schedule: PulseSchedule, qblox_instruments: list[QbloxModule]
+    ) -> tuple[list[tuple[AWGQbloxSequencer, PulseBusSchedule]], list[tuple[AWGQbloxSequencer, PulseBusSchedule]]]:
+        """Gets the pulse schedule for each sequencer. This corresponds to a PulseBusSchedule
+        object. Note that for multiplexed QRM more than one PulseBusSchedule is sent to the same
+        bus (feedline) but to different sequences (as qubit schedules)
+
+        Args:
+            pulse_schedule (PulseSchedule): pulse schedule to be executed
+            qblox_instruments (list[QbloxModule]): list of connected qblox modules
+
+        Raises:
+            ValueError: raises an error if readoud pulses are targeted at more tan one port
+
+        Returns:
+            dict[AWGQbloxSequencer, PulseBusSchedule]: dictionary of the pulse schedule corresponding to a given sequencer
+        """
+        qrm_sequencers = [
+            sequencer
+            for instrument in qblox_instruments
+            for sequencer in instrument.awg_sequencers
+            if instrument.name in self.readout_modules
+        ]
+        feedline_port = qrm_sequencers[0].chip_port_id  # FIXME: only one chip port id supported for qrm
```

Comment:
This is related to all the buses problems for feedline input currently pointing to 5 diferent sequencers. This FIXME is related to adding support for more than one QRM in the compiler (I'll add a task for that since we are supporting this in the iqm chip)